Added forbidden datasets

beccarobins opened the pull request.

richierocks commented on docs/courses/design/brainstorming-datasets.md:
```diff
@@ -71,9 +71,13 @@ Bigger datasets are not always better. In many cases, having a dataset that is s
 ## Common problems and their solutions
 
-### Using the iris dataset
+### Forbidden datasets
```
Consider adding "Titanic".

richierocks replied:
"Forbidden" sounds a little bit gratuitously harsh. It also ought to be restated as a problem (for consistency with the rest of the docs). Maybe "Using overused datasets".

beccarobins replied:
I'd like to revamp all of this actually. There are too many sections at the moment and the formatting doesn't make the division between sections super clear (the font sizes are not _that_ different). I think that "Using overused datasets" doesn't convey the idea of "Don't use these", but rather "use at your own peril".

beccarobins commented:
@richierocks I've now updated it quite a bit, but I'd like you to make an addition as well. I've added "Can I use an API?" to the FAQ section and I think you have a better grasp of how to explain to instructors how we _can't_ use an API, but we can fake an API call. It's (maybe) fair to assume that instructors wanting to use an API will understand why it's not feasible.

richierocks commented:
@beccarobins I love the content in your edits; I don't the like file structure change (since it makes it different to the other Design Step pages). I've attempted a synthesizing edit that includes your ideas but retains the old structure.

I also added in the API FAQ, and the other bits closing issues related to this page.

beccarobins commented:
@richierocks The file structure **is a problem** because there are WAY too many sections. No one in their right mind would read all this. It's a mess.
